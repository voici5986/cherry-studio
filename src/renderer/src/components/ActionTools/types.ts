/**
 * 动作工具基本信息
 */
export interface ActionToolSpec {
  id: string
  type: 'core' | 'quick'
  order: number
}

/**
 * 动作工具定义接口
 * @param id 唯一标识符
 * @param type 工具类型
 * @param order 显示顺序，越小越靠右
 * @param icon 按钮图标
 * @param tooltip 提示文本
 * @param visible 显示条件
 * @param onClick 点击动作
 * @param children 子工具（例如 more 下拉菜单）
 */
export interface ActionTool extends ActionToolSpec {
  icon: React.ReactNode
  tooltip?: string
  visible?: () => boolean
  onClick?: () => void
  children?: Omit<ActionTool, 'children'>[]
}

/**
 * 子组件向父组件注册工具所需的 props
 */
export interface ToolRegisterProps {
  setTools?: (value: React.SetStateAction<ActionTool[]>) => void
}
