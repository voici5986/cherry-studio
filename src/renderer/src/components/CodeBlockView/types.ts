/**
 * 视图模式
 */
export type ViewMode = 'source' | 'special' | 'split'
