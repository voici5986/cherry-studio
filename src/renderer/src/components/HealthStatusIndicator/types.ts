import { HealthStatus } from '@renderer/types/healthCheck'

/**
 * 用于展示单个健康检查结果的必要数据
 */
export interface HealthResult {
  status: HealthStatus
  latency?: number
  error?: string
  // 用于在 Tooltip 中显示额外上下文信息，例如 API Key 或模型名称
  label?: string
}
