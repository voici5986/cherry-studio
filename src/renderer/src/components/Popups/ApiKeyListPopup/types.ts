import { PreprocessProvider, Provider, WebSearchProvider } from '@renderer/types'

/**
 * API key 格式有效性
 */
export type ApiKeyValidity = {
  isValid: boolean
  error?: string
}

export type ApiProviderUnion = Provider | WebSearchProvider | PreprocessProvider

export type ApiProviderKind = 'llm' | 'websearch' | 'doc-preprocess'
