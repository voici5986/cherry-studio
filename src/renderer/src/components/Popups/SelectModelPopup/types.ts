import { Model } from '@renderer/types'
import { ReactNode } from 'react'

// 列表项类型，组名也作为列表项
export type ListItemType = 'group' | 'model'

// 滚动触发来源类型
export type ScrollTrigger = 'initial' | 'search' | 'keyboard' | 'none'

// 扁平化列表项接口
export interface FlatListItem {
  key: string
  type: ListItemType
  icon?: ReactNode
  name: ReactNode
  tags?: ReactNode
  model?: Model
  isPinned?: boolean
  isSelected?: boolean
}
