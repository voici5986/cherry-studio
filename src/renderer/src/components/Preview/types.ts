/**
 * 预览组件的基本 props
 */
export interface BasicPreviewProps {
  children: string
  enableToolbar?: boolean
}

/**
 * 通过 useImperativeHandle 暴露的方法类型
 */
export interface BasicPreviewHandles {
  pan: (dx: number, dy: number, absolute?: boolean) => void
  zoom: (delta: number, absolute?: boolean) => void
  copy: () => Promise<void>
  download: (format: 'svg' | 'png') => Promise<void>
}
