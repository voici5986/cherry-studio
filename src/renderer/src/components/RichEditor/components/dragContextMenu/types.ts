import type { Editor } from '@tiptap/core'
import type { Node } from '@tiptap/pm/model'
import type { ReactNode } from 'react'

/**
 * 操作组类型 - 用于菜单项分组
 */
export enum ActionGroup {
  TRANSFORM = 'transform', // 节点转换操作
  FORMAT = 'format', // 格式化操作
  BLOCK = 'block', // 块级操作
  INSERT = 'insert', // 插入操作
  AI = 'ai' // AI 相关操作 (预留)
}

/**
 * 菜单操作项接口
 */
export interface MenuAction {
  /** 操作唯一标识 */
  id: string
  /** 显示标签 */
  label: string
  /** 图标 */
  icon?: ReactNode
  /** 操作组 */
  group: ActionGroup
  /** 快捷键描述 */
  shortcut?: string
  /** 是否为危险操作 */
  danger?: boolean
  /** 是否可用 */
  isEnabled: (editor: Editor, node: Node, pos: number) => boolean
  /** 执行操作 */
  execute: (editor: Editor, node: Node, pos: number) => void
  /** 自定义类名 */
  className?: string
}

/**
 * 节点转换选项
 */
export interface NodeTransformOptions {
  /** 目标节点类型 */
  nodeType: string
  /** 节点属性 */
  attrs?: Record<string, any>
  /** 是否保留内容 */
  preserveContent?: boolean
}

/**
 * 拖拽上下文菜单属性
 */
export interface DragContextMenuProps {
  /** 编辑器实例 */
  editor: Editor
  /** 当前节点 */
  node: Node
  /** 节点在文档中的位置 */
  position: number
  /** 菜单显示状态 */
  visible: boolean
  /** 菜单位置 */
  menuPosition: { x: number; y: number }
  /** 关闭回调 */
  onClose: () => void
  /** 自定义操作 */
  customActions?: MenuAction[]
  /** 禁用的操作 ID 列表 */
  disabledActions?: string[]
}

/**
 * 拖拽手柄属性
 */
export interface DragHandleProps {
  /** 编辑器实例 */
  editor: Editor
  /** 当前节点 */
  node: Node
  /** 节点位置 */
  position: number
  /** 是否显示 */
  visible: boolean
  /** 点击回调 */
  onClick: () => void
  /** 拖拽开始回调 */
  onDragStart?: (e: DragEvent) => void
  /** 自定义类名 */
  className?: string
}

/**
 * 菜单可见性配置
 */
export interface MenuVisibilityOptions {
  /** 当前编辑器实例 */
  editor: Editor
  /** 当前节点 */
  node: Node
  /** 节点位置 */
  position: number
  /** 自定义可见性规则 */
  customRules?: Array<(editor: Editor, node: Node, pos: number) => boolean>
}

/**
 * 菜单操作结果
 */
export interface MenuActionResult {
  /** 是否成功执行 */
  success: boolean
  /** 错误信息 */
  error?: string
  /** 是否需要关闭菜单 */
  shouldCloseMenu?: boolean
}

/**
 * 颜色选择器选项
 */
export interface ColorOption {
  /** 颜色值 */
  color: string
  /** 显示名称 */
  name: string
  /** 是否为默认颜色 */
  isDefault?: boolean
}

/**
 * 节点转换映射
 */
export interface NodeTransformMap {
  [key: string]: {
    /** 显示名称 */
    label: string
    /** 图标 */
    icon: ReactNode
    /** 转换配置 */
    transform: NodeTransformOptions
    /** 是否可用的检查函数 */
    isAvailable?: (editor: Editor, currentNode: Node) => boolean
  }
}

/**
 * 拖拽上下文菜单配置
 */
export interface DragContextMenuConfig {
  /** 是否启用 */
  enabled: boolean
  /** 默认操作列表 */
  defaultActions: string[]
  /** 自定义操作 */
  customActions?: MenuAction[]
  /** 操作组排序 */
  groupOrder?: ActionGroup[]
  /** 颜色选择器配置 */
  colorOptions?: ColorOption[]
  /** 节点转换映射 */
  transformMap?: NodeTransformMap
  /** 菜单样式配置 */
  menuStyles?: {
    maxWidth?: number
    maxHeight?: number
    showShortcuts?: boolean
  }
}

/**
 * 钩子返回值 - useDragContextMenu
 */
export interface UseDragContextMenuReturn {
  /** 菜单是否可见 */
  isMenuVisible: boolean
  /** 菜单位置 */
  menuPosition: { x: number; y: number }
  /** 当前节点信息 */
  currentNode: { node: Node; position: number } | null
  /** 显示菜单 */
  showMenu: (node: Node, position: number, clientPos: { x: number; y: number }) => void
  /** 隐藏菜单 */
  hideMenu: () => void
  /** 执行操作 */
  executeAction: (action: MenuAction) => Promise<MenuActionResult>
}

/**
 * 钩子返回值 - useMenuActionVisibility
 */
export interface UseMenuActionVisibilityReturn {
  /** 可见的操作列表 */
  visibleActions: MenuAction[]
  /** 按组分类的操作 */
  actionsByGroup: Record<ActionGroup, MenuAction[]>
  /** 刷新可见性 */
  refreshVisibility: () => void
}

/**
 * 事件处理器类型
 */
export interface EventHandlers {
  onMenuShow?: (node: Node, position: number) => void
  onMenuHide?: () => void
  onActionExecute?: (action: MenuAction, result: MenuActionResult) => void
  onError?: (error: Error, context: string) => void
}

/**
 * 位置计算选项
 */
export interface PositionOptions {
  /** 偏移量 */
  offset?: { x: number; y: number }
  /** 边界约束 */
  boundary?: HTMLElement
  /** 对齐方式 */
  align?: 'start' | 'center' | 'end'
  /** 自动调整位置 */
  autoAdjust?: boolean
}
