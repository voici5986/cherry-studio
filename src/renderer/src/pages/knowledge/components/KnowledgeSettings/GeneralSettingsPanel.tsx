import InfoTooltip from '@renderer/components/InfoTooltip'
import InputEmbeddingDimension from '@renderer/components/InputEmbeddingDimension'
import ModelSelector from '@renderer/components/ModelSelector'
import { DEFAULT_KNOWLEDGE_DOCUMENT_COUNT } from '@renderer/config/constant'
import { isEmbeddingModel, isRerankModel } from '@renderer/config/models'
import { useProviders } from '@renderer/hooks/useProvider'
import { getModelUniqId } from '@renderer/services/ModelService'
import { KnowledgeBase, PreprocessProvider } from '@renderer/types'
import { Input, Select, SelectProps, Slider } from 'antd'
import { useTranslation } from 'react-i18next'

import { SettingsItem, SettingsPanel } from './styles'

interface GeneralSettingsPanelProps {
  newBase: KnowledgeBase
  setNewBase: React.Dispatch<React.SetStateAction<KnowledgeBase>>
  selectedDocPreprocessProvider?: PreprocessProvider
  docPreprocessSelectOptions: SelectProps['options']
  handlers: {
    handleEmbeddingModelChange: (value: string) => void
    handleDimensionChange: (value: number | null) => void
    handleRerankModelChange: (value: string) => void
    handleDocPreprocessChange: (value: string) => void
  }
}

const GeneralSettingsPanel: React.FC<GeneralSettingsPanelProps> = ({
  newBase,
  setNewBase,
  selectedDocPreprocessProvider,
  docPreprocessSelectOptions,
  handlers
}) => {
  const { t } = useTranslation()
  const { providers } = useProviders()
  const { handleEmbeddingModelChange, handleDimensionChange, handleRerankModelChange, handleDocPreprocessChange } =
    handlers

  return (
    <SettingsPanel>
      <SettingsItem>
        <div className="settings-label">{t('common.name')}</div>
        <Input
          placeholder={t('common.name')}
          value={newBase.name}
          onChange={(e) => setNewBase((prev) => ({ ...prev, name: e.target.value }))}
        />
      </SettingsItem>

      <SettingsItem>
        <div className="settings-label">
          {t('models.embedding_model')}
          <InfoTooltip title={t('models.embedding_model_tooltip')} placement="right" />
        </div>
        <ModelSelector
          providers={providers}
          predicate={isEmbeddingModel}
          style={{ width: '100%' }}
          placeholder={t('settings.models.empty')}
          value={getModelUniqId(newBase.model)}
          onChange={handleEmbeddingModelChange}
        />
      </SettingsItem>

      <SettingsItem>
        <div className="settings-label">
          {t('knowledge.dimensions')}
          <InfoTooltip title={t('knowledge.dimensions_size_tooltip')} placement="right" />
        </div>
        <InputEmbeddingDimension
          value={newBase.dimensions}
          onChange={handleDimensionChange}
          model={newBase.model}
          disabled={!newBase.model}
        />
      </SettingsItem>

      <SettingsItem>
        <div className="settings-label">
          {t('models.rerank_model')}
          <InfoTooltip title={t('models.rerank_model_tooltip')} placement="right" />
        </div>
        <ModelSelector
          providers={providers}
          predicate={isRerankModel}
          style={{ width: '100%' }}
          value={getModelUniqId(newBase.rerankModel) || undefined}
          placeholder={t('settings.models.empty')}
          onChange={handleRerankModelChange}
          allowClear
        />
      </SettingsItem>

      <SettingsItem>
        <div className="settings-label">
          {t('settings.tool.preprocess.title')}
          <InfoTooltip title={t('settings.tool.preprocess.tooltip')} placement="right" />
        </div>
        <Select
          value={selectedDocPreprocessProvider?.id}
          style={{ width: '100%' }}
          onChange={handleDocPreprocessChange}
          placeholder={t('settings.tool.preprocess.provider_placeholder')}
          options={docPreprocessSelectOptions}
          allowClear
        />
      </SettingsItem>

      <SettingsItem>
        <div className="settings-label">
          {t('knowledge.document_count')}
          <InfoTooltip title={t('knowledge.document_count_help')} placement="right" />
        </div>
        <Slider
          style={{ width: '100%' }}
          min={1}
          max={50}
          step={1}
          value={newBase.documentCount || DEFAULT_KNOWLEDGE_DOCUMENT_COUNT}
          marks={{ 1: '1', 6: t('knowledge.document_count_default'), 30: '30', 50: '50' }}
          onChange={(value) => setNewBase((prev) => ({ ...prev, documentCount: value }))}
        />
      </SettingsItem>
    </SettingsPanel>
  )
}

export default GeneralSettingsPanel
