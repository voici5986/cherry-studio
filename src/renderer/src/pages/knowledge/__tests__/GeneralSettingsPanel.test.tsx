import { KnowledgeBase, Model, PreprocessProvider } from '@renderer/types'
import { fireEvent, render, screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'
import { beforeEach, describe, expect, it, vi } from 'vitest'

import GeneralSettingsPanel from '../components/KnowledgeSettings/GeneralSettingsPanel'

// Mock dependencies
const mocks = vi.hoisted(() => ({
  t: vi.fn((key: string) => key),
  providers: [
    {
      id: 'openai',
      name: 'OpenAI',
      models: [
        {
          id: 'text-embedding-3-small',
          provider: 'openai',
          name: 'text-embedding-3-small',
          group: 'embedding'
        }
      ]
    }
  ],
  handlers: {
    handleEmbeddingModelChange: vi.fn(),
    handleDimensionChange: vi.fn(),
    handleRerankModelChange: vi.fn(),
    handleDocPreprocessChange: vi.fn()
  }
}))

// Mock InfoTooltip component
vi.mock('@renderer/components/InfoTooltip', () => ({
  default: ({ title, placement }: { title: string; placement: string }) => (
    <span data-testid="info-tooltip" title={title} data-placement={placement}>
      ℹ️
    </span>
  )
}))

// Mock ModelSelector component
vi.mock('@renderer/components/ModelSelector', () => ({
  default: ({ value, onChange, placeholder, allowClear, providers, predicate }: any) => {
    // Determine if this is for embedding or rerank models based on predicate
    const isEmbedding = predicate?.toString().includes('embedding')
    const isRerank = predicate?.toString().includes('rerank')

    // Use providers parameter to avoid lint error
    const hasProviders = providers && providers.length > 0

    return (
      <select
        data-testid="model-selector"
        value={value || ''}
        onChange={(e) => onChange?.(e.target.value)}
        data-placeholder={placeholder}
        data-allow-clear={allowClear}
        data-model-type={isEmbedding ? 'embedding' : isRerank ? 'rerank' : 'unknown'}
        data-has-providers={hasProviders}>
        <option value="">Select model</option>
        {isEmbedding && (
          <>
            <option value="openai/text-embedding-3-small">text-embedding-3-small</option>
            <option value="openai/text-embedding-ada-002">text-embedding-ada-002</option>
          </>
        )}
        {isRerank && (
          <>
            <option value="openai/rerank-model">rerank-model</option>
            <option value="cohere/rerank-english-v2.0">rerank-english-v2.0</option>
          </>
        )}
      </select>
    )
  }
}))

// Mock InputEmbeddingDimension component
vi.mock('@renderer/components/InputEmbeddingDimension', () => ({
  default: ({ value, onChange, model, disabled }: any) => (
    <input
      data-testid="embedding-dimension-input"
      type="number"
      value={value || ''}
      onChange={(e) => onChange?.(Number(e.target.value))}
      disabled={disabled}
      data-model={model?.id}
    />
  )
}))

// Mock useProviders hook
vi.mock('@renderer/hooks/useProvider', () => ({
  useProviders: () => ({ providers: mocks.providers })
}))

// Mock ModelService
vi.mock('@renderer/services/ModelService', () => ({
  getModelUniqId: (model: Model | undefined) => (model ? `${model.provider}/${model.id}` : undefined)
}))

// Mock model predicates
vi.mock('@renderer/config/models', () => ({
  isEmbeddingModel: (model: Model) => model.group === 'embedding',
  isRerankModel: (model: Model) => model.group === 'rerank'
}))

// Mock constant
vi.mock('@renderer/config/constant', () => ({
  DEFAULT_KNOWLEDGE_DOCUMENT_COUNT: 6
}))

// Mock react-i18next
vi.mock('react-i18next', () => ({
  useTranslation: () => ({ t: mocks.t })
}))

// Mock antd components
vi.mock('antd', () => ({
  Input: ({ value, onChange, placeholder }: any) => (
    <input data-testid="name-input" value={value} onChange={onChange} placeholder={placeholder} />
  ),
  Select: ({ value, onChange, placeholder, options, allowClear, children }: any) => (
    <select
      data-testid="preprocess-select"
      value={value || ''}
      onChange={(e) => onChange?.(e.target.value)}
      data-placeholder={placeholder}
      data-allow-clear={allowClear}>
      <option value="">Select option</option>
      {options?.map((option: any) => (
        <option key={option.value} value={option.value}>
          {option.label}
        </option>
      ))}
      {children}
    </select>
  ),
  Slider: ({ value, onChange, min, max, step, marks }: any) => (
    <input
      data-testid="document-count-slider"
      type="range"
      value={value}
      onChange={(e) => onChange?.(Number(e.target.value))}
      min={min}
      max={max}
      step={step}
      data-marks={JSON.stringify(marks)}
    />
  )
}))

/**
 * 创建测试用的 KnowledgeBase 对象
 * @param overrides - 可选的属性覆盖
 * @returns 完整的 KnowledgeBase 对象
 */
function createKnowledgeBase(overrides: Partial<KnowledgeBase> = {}): KnowledgeBase {
  const defaultModel: Model = {
    id: 'text-embedding-3-small',
    provider: 'openai',
    name: 'text-embedding-3-small',
    group: 'embedding'
  }

  return {
    id: 'test-base-id',
    name: 'Test Knowledge Base',
    model: defaultModel,
    items: [],
    created_at: Date.now(),
    updated_at: Date.now(),
    version: 1,
    ...overrides
  }
}

/**
 * 创建测试用的 PreprocessProvider 对象
 * @param overrides - 可选的属性覆盖
 * @returns 完整的 PreprocessProvider 对象
 */
function createPreprocessProvider(overrides: Partial<PreprocessProvider> = {}): PreprocessProvider {
  return {
    id: 'doc2x',
    name: 'Doc2X',
    apiKey: 'test-api-key',
    ...overrides
  }
}

describe('GeneralSettingsPanel', () => {
  const mockBase = createKnowledgeBase()
  const mockSetNewBase = vi.fn()
  const mockSelectedDocPreprocessProvider = createPreprocessProvider()
  const mockDocPreprocessSelectOptions = [
    { value: 'doc2x', label: 'Doc2X' },
    { value: 'mistral', label: 'Mistral' }
  ]

  // 提取公共渲染函数
  const renderComponent = (props: Partial<any> = {}) => {
    return render(
      <GeneralSettingsPanel
        newBase={mockBase}
        setNewBase={mockSetNewBase}
        selectedDocPreprocessProvider={mockSelectedDocPreprocessProvider}
        docPreprocessSelectOptions={mockDocPreprocessSelectOptions}
        handlers={mocks.handlers}
        {...props}
      />
    )
  }

  beforeEach(() => {
    vi.clearAllMocks()
  })

  describe('basic rendering', () => {
    it('should match snapshot', () => {
      const { container } = renderComponent()
      expect(container.firstChild).toMatchSnapshot()
    })

    it('should render without selectedDocPreprocessProvider', () => {
      renderComponent({ selectedDocPreprocessProvider: undefined })
      expect(screen.getByTestId('preprocess-select')).toHaveValue('')
    })

    it('should render with empty docPreprocessSelectOptions', () => {
      renderComponent({ docPreprocessSelectOptions: [] })
      const preprocessSelect = screen.getByTestId('preprocess-select')
      expect(preprocessSelect.children).toHaveLength(1)
    })
  })

  describe('functionality', () => {
    const user = userEvent.setup()

    it('should handle name input change', async () => {
      renderComponent()

      const nameInput = screen.getByTestId('name-input')
      await user.type(nameInput, 'New Knowledge Base Name')

      expect(mockSetNewBase).toHaveBeenCalledWith(expect.any(Function))
    })

    it('should handle preprocess provider change', async () => {
      renderComponent()

      const preprocessSelect = screen.getByTestId('preprocess-select')
      await user.selectOptions(preprocessSelect, 'mistral')

      expect(mocks.handlers.handleDocPreprocessChange).toHaveBeenCalledWith('mistral')
    })

    it('should handle model selection changes', async () => {
      renderComponent()

      const modelSelectors = screen.getAllByTestId('model-selector')

      // Test embedding model change
      const embeddingModelSelector = modelSelectors[0]
      await user.selectOptions(embeddingModelSelector, 'openai/text-embedding-ada-002')
      expect(mocks.handlers.handleEmbeddingModelChange).toHaveBeenCalledWith('openai/text-embedding-ada-002')

      // Test rerank model change
      const rerankModelSelector = modelSelectors[1]
      await user.selectOptions(rerankModelSelector, 'openai/rerank-model')
      expect(mocks.handlers.handleRerankModelChange).toHaveBeenCalledWith('openai/rerank-model')
    })

    it('should handle dimension change', async () => {
      renderComponent()

      const dimensionInput = screen.getByTestId('embedding-dimension-input')
      fireEvent.change(dimensionInput, { target: { value: '1536' } })

      expect(mocks.handlers.handleDimensionChange).toHaveBeenCalledWith(1536)
    })

    it('should handle document count change', async () => {
      renderComponent()

      const documentCountSlider = screen.getByTestId('document-count-slider')
      fireEvent.change(documentCountSlider, { target: { value: '10' } })

      expect(mockSetNewBase).toHaveBeenCalledWith(expect.any(Function))
    })

    it('should disable dimension input when no model is selected', () => {
      const baseWithoutModel = createKnowledgeBase({ model: undefined as any })
      renderComponent({ newBase: baseWithoutModel })

      const dimensionInput = screen.getByTestId('embedding-dimension-input')
      expect(dimensionInput).toBeDisabled()
    })
  })
})
