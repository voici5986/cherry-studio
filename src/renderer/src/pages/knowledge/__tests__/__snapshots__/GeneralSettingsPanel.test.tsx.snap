// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GeneralSettingsPanel > basic rendering > should match snapshot 1`] = `
.c0 {
  padding: 0 16px;
}

.c1 {
  margin-bottom: 24px;
}

.c1 .settings-label {
  font-size: 14px;
  margin-bottom: 8px;
  display: flex;
  align-items: center;
  gap: 8px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="settings-label"
    >
      common.name
    </div>
    <input
      data-testid="name-input"
      placeholder="common.name"
      value="Test Knowledge Base"
    />
  </div>
  <div
    class="c1"
  >
    <div
      class="settings-label"
    >
      models.embedding_model
      <span
        data-placement="right"
        data-testid="info-tooltip"
        title="models.embedding_model_tooltip"
      >
        ℹ️
      </span>
    </div>
    <select
      data-has-providers="true"
      data-model-type="embedding"
      data-placeholder="settings.models.empty"
      data-testid="model-selector"
    >
      <option
        value=""
      >
        Select model
      </option>
      <option
        value="openai/text-embedding-3-small"
      >
        text-embedding-3-small
      </option>
      <option
        value="openai/text-embedding-ada-002"
      >
        text-embedding-ada-002
      </option>
    </select>
  </div>
  <div
    class="c1"
  >
    <div
      class="settings-label"
    >
      knowledge.dimensions
      <span
        data-placement="right"
        data-testid="info-tooltip"
        title="knowledge.dimensions_size_tooltip"
      >
        ℹ️
      </span>
    </div>
    <input
      data-model="text-embedding-3-small"
      data-testid="embedding-dimension-input"
      type="number"
      value=""
    />
  </div>
  <div
    class="c1"
  >
    <div
      class="settings-label"
    >
      models.rerank_model
      <span
        data-placement="right"
        data-testid="info-tooltip"
        title="models.rerank_model_tooltip"
      >
        ℹ️
      </span>
    </div>
    <select
      data-allow-clear="true"
      data-has-providers="true"
      data-model-type="rerank"
      data-placeholder="settings.models.empty"
      data-testid="model-selector"
    >
      <option
        value=""
      >
        Select model
      </option>
      <option
        value="openai/rerank-model"
      >
        rerank-model
      </option>
      <option
        value="cohere/rerank-english-v2.0"
      >
        rerank-english-v2.0
      </option>
    </select>
  </div>
  <div
    class="c1"
  >
    <div
      class="settings-label"
    >
      settings.tool.preprocess.title
      <span
        data-placement="right"
        data-testid="info-tooltip"
        title="settings.tool.preprocess.tooltip"
      >
        ℹ️
      </span>
    </div>
    <select
      data-allow-clear="true"
      data-placeholder="settings.tool.preprocess.provider_placeholder"
      data-testid="preprocess-select"
    >
      <option
        value=""
      >
        Select option
      </option>
      <option
        value="doc2x"
      >
        Doc2X
      </option>
      <option
        value="mistral"
      >
        Mistral
      </option>
    </select>
  </div>
  <div
    class="c1"
  >
    <div
      class="settings-label"
    >
      knowledge.document_count
      <span
        data-placement="right"
        data-testid="info-tooltip"
        title="knowledge.document_count_help"
      >
        ℹ️
      </span>
    </div>
    <input
      data-marks="{"1":"1","6":"knowledge.document_count_default","30":"30","50":"50"}"
      data-testid="document-count-slider"
      max="50"
      min="1"
      step="1"
      type="range"
      value="6"
    />
  </div>
</div>
`;
